### PR TITLE
Increase spire-agent healthcheck wait time

### PR DESCRIPTION
**Pull request by Bolodya1997**
Spire agent fails to start in 10 attempts, need to add extra tumeout.

### CODE REVIEW

**Review by denis-tingajkin — COMMENTED** [review]
I would also recommend adding a check that if the context has no deadline or context cannot be canceld (for example context.Background) then we need to use some default timeout.

File: pkg/tools/spire/start.go [hunk 1]
```diff
@@ -220,3 +209,23 @@ func writeDefaultConfigFiles(ctx context.Context, spireRoot string) (string, err
...
 }
+
+func execHealthCheck(ctx context.Context, cmdStr string, options ...*exechelper.Option) error {
+	if _, ok := ctx.Deadline(); !ok {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, healthCheckTimeout)
+		defer cancel()
+	}
+
+healthCheck:
```
Comment (denis-tingajkin) [inline pkg/tools/spire/start.go:220]: Can we do not use labels? I would suggest just use a for loop
```suggestion:
for {
```

File: pkg/tools/spire/start.go [hunk 2]
```diff
@@ -220,3 +209,22 @@ func writeDefaultConfigFiles(ctx context.Context, spireRoot string) (string, err
...
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, healthCheckTimeout)
+		defer cancel()
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
```
Comment (denis-tingajkin) [inline pkg/tools/spire/start.go:224]: It looks like this branch could load CPU. Should we add wait in case if err != nil?

Comment (Bolodya1997) [reply pkg/tools/spire/start.go:224]: `exechelper.Run(...)` starts process with `os/exec` and waits for its completion, so there is actually a wait, it is not a "for true" case.

Comment (denis-tingajkin) [reply pkg/tools/spire/start.go:224]: yes, but the process may fail in 50 ms

Comment (Bolodya1997) [reply pkg/tools/spire/start.go:224]: We can, so why not :)

**Comment by edwarnicke** [comment]
This PR appears to be breaking downstream: 

https://github.com/networkservicemesh/cmd-forwarder-vppagent/pull/156/checks?check_run_id=1179132829

Also... I'm not clear on quite what the issue being solved here was...

**Pull request merged**